src/cli/backup/teams.go

```diff
@@ -217,14 +254,55 @@ func validateTeamBackupCreateFlags(teams []string) error {
 		)
 	}
 
-	// TODO(meain)
-	// for _, d := range cats {
-	// 	if d != dataLibraries {
-	// 		return clues.New(
-	// 			d + " is an unrecognized data type; only  " + dataLibraries + " is supported"
-	// 		)
-	// 	}
-	// }
+	msg := fmt.Sprintf(
+		" is an unrecognized data type; only %s and %s are supported",
+		dataLibraries, dataMessages)
+
+	allowedCats := map[string]struct{}{
+		dataLibraries: {},
+		dataMessages:  {},
+	}
+
+	for _, d := range cats {
+		if _, ok := allowedCats[d]; !ok {
+			return clues.New(d + msg)
+		}
+	}
 
 	return nil
 }
+
+func teamsBackupCreateSelectors(
+	ctx context.Context,
+	ins idname.Cacher,
+	team, cats []string,
+) *selectors.GroupsBackup {
+	if filters.PathContains(team).Compare(flags.Wildcard) {
+		return includeAllTeamWithCategories(ins, cats)
+	}
+
+	sel := selectors.NewGroupsBackup(slices.Clone(team))
+
+	return addTeamsCategories(sel, cats)
+}
+
+func includeAllTeamWithCategories(ins idname.Cacher, categories []string) *selectors.GroupsBackup {
+	return addTeamsCategories(selectors.NewGroupsBackup(ins.IDs()), categories)
+}
+
+func addTeamsCategories(sel *selectors.GroupsBackup, cats []string) *selectors.GroupsBackup {
```

Comment:
could you please add a TODO stating that this should be deduped with the groups code?

Reply:
moved it to utils to dedup